packages/access-client/src/agent.js

```diff
@@ -417,9 +674,16 @@ export class Agent {
    * @param {string} email
    * @param {object} [opts]
    * @param {AbortSignal} [opts.signal]
+   * @param {Ucanto.DID<'key'>} [opts.space] - space to register
+   * @param {Ucanto.DID<'web'>} [opts.provider] - provider to register - defaults to this.connection.id
    */
-  async registerSpace(email, opts) {
-    const space = this.currentSpace()
+  async registerSpace(email, opts = {}) {
+    // if the client passes `provider` use the new space registration flow
+    if (opts.provider) {
+      return this.#newRegisterSpace(email, opts)
```

Comment:
Irakli had proposed not doing new functionality here via option.
I'd be happy if we just made the new stuff usable via 'use case' fn(agent, ...)